### PR TITLE
137 amount value shortener

### PR DESCRIPTION
**Pull request by mindfulme**
#### What's this PR do?
Adds Amount Value Shortener function along with two unit tests (i.e. for millions and for thousands)
#### Where should the reviewer start?
The function is placed in helpers/utils.js, and named `amountValueShortener`
The unit tests can be found in the appropriate folder used for unit tests
#### How should this be manually tested?
In the browser check out Markets component on the appropriate page
or
`npm run test:unit`
#### Any background context you want to provide?
The value shortener can be improved even more in just several minutes to accept even bigger values (i.e. trillions and much much more)
#### What are the relevant tickets?
#137 
#### Screenshots (if appropriate)
screen shot 2018-11-01 at 20 07 12

#### URL
Provide a link to your deploy from http://staging-ui.trusty.apasia.tech:8080/
http://staging-ui.trusty.apasia.tech:8080/137-amount-value-shortener

#### Questions:
-

### CODE REVIEW

**Comment by mindfulme** [comment]
@roma219 I can quickly improve it to parse much bigger amounts to billions, and trillions ...

File: src/helpers/utils.js [hunk 1]
```diff
@@ -2,3 +2,19 @@
...
 // convert sum to currency format, exapmle: 1 000 000 (use in markets)
 export const getVolumeFormat = sum => sum.toString().replace(/(\d)(?=(\d\d\d)+([^\d]|$))/g, '$1 ')
+// converts amount to shortened format
+export const amountValueShortener = amount => {
+  let data = [
+    { value: 1, symbol: '' },
+    { value: 1E3, symbol: 'k' },
+    { value: 1E6, symbol: 'm' }
+  ]
+  let regex = /\.0+$|(\.[0-9]*[1-9])0+$/
```
Comment (roma219) [inline src/helpers/utils.js:12]: what does this regex mean?

File: src/helpers/utils.js [hunk 2]
```diff
@@ -2,3 +2,19 @@
...
+  let i
+  for (i = data.length - 1; i > 0; i--) {
+    if (amount >= data[i].value) {
+      break
+    }
+  }
+  return (amount / data[i].value).toFixed(1).replace(regex, '$1') + data[i].symbol
```
Comment (roma219) [inline src/helpers/utils.js]: what is $1 and how does this expression work?

**Review by roma219 — CHANGES_REQUESTED** [review]
Needs some explanations

File: tests/unit/amountValueShortener.spec.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { amountValueShortener } from '@/helpers/utils'
+
+describe('amountValueShortener function', () => {
```
Comment (roma219) [inline tests/unit/amountValueShortener.spec.js:3]: Also test with some less that 1k number

File: src/helpers/utils.js (hunk 1 above)
Comment (mindfulme) [reply src/helpers/utils.js:12]: regular expression

File: tests/unit/amountValueShortener.spec.js (hunk 3 above)
Comment (mindfulme) [reply tests/unit/amountValueShortener.spec.js:3]: okay, will be done

File: src/helpers/utils.js (hunk 2 above)
Comment (mindfulme) [reply src/helpers/utils.js]: I will also write a test for it

File: src/helpers/utils.js (hunk 1 above)
Comment (mindfulme) [reply src/helpers/utils.js:12]: it matches digits

File: src/helpers/utils.js (hunk 2 above)
Comment (mindfulme) [reply src/helpers/utils.js]: or did you mean the code explanation?

Comment (mindfulme) [reply src/helpers/utils.js]: if so than it is used for substring match, here it matches for the first substring

Comment (roma219) [reply src/helpers/utils.js]: what is  `'$1'` in this line of code?

File: src/helpers/utils.js (hunk 1 above)
Comment (roma219) [reply src/helpers/utils.js:12]: I'am asking what exactly does it match? why there is `[0-9]` followed by `*[1-9]`?

File: src/helpers/utils.js [hunk 4]
```diff
@@ -2,3 +2,19 @@
 
 // convert sum to currency format, exapmle: 1 000 000 (use in markets)
 export const getVolumeFormat = sum => sum.toString().replace(/(\d)(?=(\d\d\d)+([^\d]|$))/g, '$1 ')
+// converts amount to shortened format
+export const amountValueShortener = amount => {
+  let data = [
+    { value: 1, symbol: '' },
+    { value: 1E3, symbol: 'k' },
```
Comment (roma219) [inline src/helpers/utils.js:9]: what does 1E3 mean?

File: src/helpers/utils.js [hunk 5]
```diff
@@ -2,3 +2,19 @@
 
 // convert sum to currency format, exapmle: 1 000 000 (use in markets)
 export const getVolumeFormat = sum => sum.toString().replace(/(\d)(?=(\d\d\d)+([^\d]|$))/g, '$1 ')
+// converts amount to shortened format
+export const amountValueShortener = amount => {
+  let data = [
```
Comment (roma219) [inline src/helpers/utils.js]: why `data`? change it to something more clear

**Review by roma219 — CHANGES_REQUESTED** [review]
Needs explaining

File: tests/unit/amountValueShortener.spec.js [hunk 6]
```diff
@@ -1,6 +1,13 @@
 import { amountValueShortener } from '@/helpers/utils'
 
 describe('amountValueShortener function', () => {
+  it('Test amount shortener used with 100', () => {
+    const amount = '100'
```
Comment (mindfulme) [inline tests/unit/amountValueShortener.spec.js:5]: @roma219 do you expect 0.1k or 100 as it is less than a thousand?

File: src/helpers/utils.js (hunk 5 above)
Comment (mindfulme) [reply src/helpers/utils.js]: okay, valuesToSymbols?

File: src/helpers/utils.js (hunk 1 above)
Comment (mindfulme) [reply src/helpers/utils.js:12]: [0-9] is used to find all digit characters

Comment (mindfulme) [reply src/helpers/utils.js:12]: `*[1-9]` matches 1-9 with more than 1 occurrence of 0-9 before

File: src/helpers/utils.js (hunk 2 above)
Comment (mindfulme) [reply src/helpers/utils.js]: selects the first substring removing every other substring that comes after

File: src/helpers/utils.js (hunk 4 above)
Comment (mindfulme) [reply src/helpers/utils.js:9]: 10^3

Comment (mindfulme) [reply src/helpers/utils.js:9]: 10 to the power of 3

File: src/helpers/utils.js (hunk 2 above)
Comment (mindfulme) [reply src/helpers/utils.js]: > what is `'$1'` in this line of code?

using it with .replace(regex, '$1') allows to trim unnecessary zeros like 199.10 to 199.1

File: src/helpers/utils.js (hunk 1 above)
Comment (mindfulme) [reply src/helpers/utils.js:12]: man, it is a really simple regex

File: tests/unit/amountValueShortener.spec.js (hunk 6 above)
Comment (mindfulme) [reply tests/unit/amountValueShortener.spec.js:5]: I leave it as 100 rather than 0.1, @roma219 if ok - merge, please

**Pull request merged**